cli/medperf/comms/entity_resources/utils.py

```diff
@@ -29,15 +32,77 @@ def __parse_resource(resource: str):
     raise InvalidArgumentError(msg)
 
 
-def download_resource(resource: str, output_path: str):
-    """Download a resource
+def valid_file_exists(output_path, expected_hash):
+    """Checks if the existing file matches the passed expected hash.
+    If it is not, or no hash was passed, it will be removed
+    from the filesystem."""
+
+    if expected_hash:
+        file_hash = get_file_sha1(output_path)
+        if file_hash == expected_hash:
+            return True
+
+    remove_path(output_path)
+    return False
+
+
+def tmp_download_resource(resource):
+    """Downloads a resource to the temporary storage."""
```

Comment:
Can you describe what exactly is being passed for `resource`? I wouldn't know how to use this function at the current stage